File: backend/test_websocket.py
import asyncio
import websockets
import json

async def test_websocket():
    """Simple test script to verify the websocket functionality"""
    uri = "ws://localhost:8000/ws/query"
    
    print("Connecting to WebSocket server...")
    async with websockets.connect(uri) as websocket:
        print("Connected to WebSocket server")
        
        # Create a test query
        test_query = {
            "subreddit": "python",
            "keyword": "asyncio",
            "question": "What are the best practices for using asyncio in Python?",
            "limit": 5,
            "repeatHours": 0,
            "repeatMinutes": 0
        }
        
        # Send the query to the server
        print(f"Sending query: {test_query}")
        await websocket.send(json.dumps(test_query))
        
        # Receive and print all messages until the connection is closed
        try:
            while True:
                response = await websocket.recv()
                data = json.loads(response)
                
                if "status" in data:
                    print(f"Status update: {data['status']}")
                    
                    # If we got final results, print them and exit
                    if "results" in data:
                        print("\nFinal results:")
                        print(json.dumps(data["results"], indent=2))
                        break
                        
                elif "error" in data:
                    print(f"Error: {data['error']}")
                    break
                    
        except websockets.exceptions.ConnectionClosed:
            print("Connection closed")

if __name__ == "__main__":
    asyncio.run(test_websocket()) 

File: frontend/app/test-websocket.js
/**
 * Simple WebSocket test client for browser console testing
 * To use:
 * 1. Open your browser console on the frontend app
 * 2. Copy and paste this entire file into the console
 * 3. Call testWebSocket() to run the test
 */

function testWebSocket() {
  console.log("Starting WebSocket test...");
  
  // Create a new WebSocket connection
  const ws = new WebSocket('ws://localhost:8000/ws/query');
  
  // Connection opened
  ws.onopen = (event) => {
    console.log("Connected to WebSocket server");
    
    // Send a test query
    const testQuery = {
      subreddit: "python",
      keyword: "asyncio",
      question: "What are the best practices for using asyncio in Python?",
      limit: 5,
      repeatHours: 0,
      repeatMinutes: 0
    };
    
    console.log("Sending test query:", testQuery);
    ws.send(JSON.stringify(testQuery));
  };
  
  // Listen for messages
  ws.onmessage = (event) => {
    try {
      const data = JSON.parse(event.data);
      console.log("Received message:", data);
      
      if (data.status) {
        console.log("Status update:", data.status);
      }
      
      if (data.results) {
        console.log("Final results received:", data.results);
      }
      
      if (data.error) {
        console.error("Error from server:", data.error);
      }
    } catch (error) {
      console.error("Error parsing message:", error);
      console.log("Raw message:", event.data);
    }
  };
  
  // Connection closed
  ws.onclose = (event) => {
    if (event.wasClean) {
      console.log(`Connection closed cleanly, code=${event.code}, reason=${event.reason}`);
    } else {
      console.error('Connection died');
    }
  };
  
  // Connection error
  ws.onerror = (error) => {
    console.error("WebSocket error:", error);
  };
  
  return ws; // Return the WebSocket instance so it can be manipulated later
}

console.log("WebSocket test client loaded. Type testWebSocket() to run the test."); 